### PR TITLE
Fix meta would send empty raft snapshot

### PR DESCRIPTION
**Pull request by critical27**
In poc project, 3 replica meta. If we remove the data dir of one. It would failed to start because it won't get the clusterId. It turned out that the leader send an empty snapshot which have nothing it. (Because it only scan the data with prefix space + part, however nothing start with them in meta)

### CODE REVIEW

**Comment by dangleptr** [comment]
My bad,  forget the meta server....

Besides that,  we should open the rocksdb WAL on metad.  So in most cases,  it will not need to catch up the data. 

WDYT?

File: src/kvstore/NebulaStore.cpp [hunk 1]
```diff
@@ -39,7 +39,9 @@ bool NebulaStore::init() {
     LOG(INFO) << "Start the raft service...";
     bgWorkers_ = std::make_shared<thread::GenericThreadPool>();
     bgWorkers_->start(FLAGS_num_workers, "nebula-bgworkers");
-    snapshot_.reset(new SnapshotManagerImpl(this));
+    if (options_.snapManBuilder_) {
```
Comment (dangleptr) [inline src/kvstore/NebulaStore.cpp]: Em....
Any better idea here?

**Comment by critical27** [comment]
> My bad, forget the meta server....
> 
> Besides that, we should open the rocksdb WAL on metad. So in most cases, it will not need to catch up the data.
> 
> WDYT?

SGTM

File: src/kvstore/NebulaStore.cpp (hunk 1 above)
Comment (critical27) [reply src/kvstore/NebulaStore.cpp]: Add a `gflag` in SnapshotManagerImpl? That would make this PR very small.

Comment (dangleptr) [reply src/kvstore/NebulaStore.cpp]: I think the key problem is we don't know the key format outside in kvstore.  So the prefix generator should be defined outside.

Comment (critical27) [reply src/kvstore/NebulaStore.cpp]: But we still to pass a lambda to NebulaStore, something like:
```
    [] (spaceId, partId) -> std::unique_ptr<KVIterator> iter {
        std::unique_ptr<KVIterator> iter;
        auto ret = store_->prefix(spaceId, partId, prefix, &iter);
        return iter;
    }
```

The KVIterator is what we need here, it seems not a big difference.

Comment (dangleptr) [reply src/kvstore/NebulaStore.cpp]: The difference is for kvstore,  it should not know anything about meta or data.

File: src/kvstore/SnapshotManagerImpl.cpp [hunk 2]
```diff
@@ -17,7 +17,7 @@ void SnapshotManagerImpl::accessAllRowsInSnapshot(GraphSpaceID spaceId,
                                                   raftex::SnapshotCallback cb) {
     CHECK_NOTNULL(store_);
     std::unique_ptr<KVIterator> iter;
-    auto prefix = NebulaKeyUtils::prefix(partId);
+    auto prefix = NebulaKeyUtils::snapshotPrefix(partId);
```
Comment (dangleptr) [inline src/kvstore/SnapshotManagerImpl.cpp]: If the prefix is empty,  what's the behavior for seek of rocksdb.

Comment (critical27) [reply src/kvstore/SnapshotManagerImpl.cpp]: Just the same as SeekToFirst and iterate over all.

File: etc/nebula-metad.conf.production [hunk 3]
```diff
@@ -33,3 +33,7 @@
 ########## storage ##########
 # Root data path, here should be only single path for metad
 --data_path=data/meta
+
+############## rocksdb Options ##############
+--rocksdb_disable_wal=false
```
Comment (Shylock-Hg) [inline etc/nebula-metad.conf.production:38]: Why not enable for storage?I think there is secure problem to loss data without replica (or with replica but all node crash).

**Pull request merged**